streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableKTableAbstractJoin.java

```diff
@@ -26,18 +29,21 @@ public abstract class KTableKTableAbstractJoin<K, V1, V2, VOut> implements
     final KTableValueGetterSupplier<K, V1> valueGetterSupplier1;
     final KTableValueGetterSupplier<K, V2> valueGetterSupplier2;
     final ValueJoiner<? super V1, ? super V2, ? extends VOut> joiner;
+    private final Set<StoreBuilder<?>> stores;
```

Comment:
Another reason I'd prefer to leave the stores unwrapped as individual StoreFactory instances rather than putting them all into a Set<StoreBuilder> before this point is because it's hard to verify that we're using the right stores here.

Is this one store or two? Is it the "this" or "other" side of the join? This'll be especially important to keep separated if we ever add further optimizations which require extra stores for table-table joins, like we have for stream-stream joins (see [here](https://github.com/apache/kafka/pull/18111/files#diff-6ca18143cc0226e6d1e4d5180ff81596a72d53639ca5184bf1238350265382a6R97-R99) for example)